docs/proposals/koordlet/20250425-heterogeneous-GPU-device-reporting.md

```diff
@@ -0,0 +1,220 @@
+  - The topology infos from directory will be synced to the annotation of the Device.
+  - The status from directory will be synced to the "Healthy" condition of the Device.
+  - The vendor and model from directory will be synced to the labels of the Device.
+```yaml
+apiVersion: scheduling.koordinator.sh/v1alpha1
+kind: Device
+metadata:
+  labels:
+    # Existing label
+    node.koordinator.sh/gpu-partition-policy: "Honor"
+    node.koordinator.sh/gpu-model: Tesla-T4 # Ascend910B/Ascend310
+    # New label
+    node.koordinator.sh/gpu-vendor: nvidia # huawei/cambricon/
+  annotations:
+    # The topology from device infos will be synced to the annotation of the Device.
+    scheduling.koordinator.sh/gpu-partitions: |
+      {
+        "1": [
+                {
+                  "minors": [
+                      0
+                  ],
+                  "gpuLinkType": "NVLink",
+                  "ringBusBandwidth": "400Gi",
+                  "allocationScore": "1"
+                }
+        ],
+        "4": [
+                {
+                    "minors": [
+                       0,1,2,3 
+                    ],
+                    # This is the topology of the Ascend910B.
+                    # https://www.hiascend.com/doc_center/source/zh/mindx-dl/50rc1/ref/dl_affinity_0002.html
+                    "gpuLinkType": "HCCS",
+                    "allocationScore": "1"
+                }
+        ]
+      }
+  name: node-1
+spec:
+  devices:
+  - health: true
+    id: GPU-fd971b33-4891-fd2e-ed42-ce6adf324615
+    minor: 0
+    resources:
+      koordinator.sh/gpu-core: "100"
+      koordinator.sh/gpu-memory: 15Gi
+      koordinator.sh/gpu-memory-ratio: "100"
+    topology:
+      busID: 0000:3b:00.0
+      nodeID: 0
+      pcieID: pci0000:3a
+      socketID: -1
+    type: gpu
+    # Add Healthy condition
+    conditions:
+    - lastTransitionTime: "2025-04-25T10:00:00Z"
+      status: "True"
+      type: Healthy
+  - health: true
+    id: GPU-0ca35a5f-551e-91e4-f3a3-c9b103afe1db
+    minor: 1
+    resources:
+      koordinator.sh/gpu-core: "100"
+      koordinator.sh/gpu-memory: 15Gi
+      koordinator.sh/gpu-memory-ratio: "100"
+    topology:
+      busID: 0000:3b:01.0
+      nodeID: 0
+      pcieID: pci0000:3a
+      socketID: -1
+    type: gpu
+    # Add Healthy condition
+    conditions:
+    - lastTransitionTime: "2025-04-25T10:00:00Z"
+      message: "XID 74: Nvlink Error"
+      reason: "NvlinkError"
+      status: "False"
+      type: Healthy
+
+```
+
+- Add the featuregate to enable the heterogeneous GPU device reporting. It is not recommended to enable both feature gates at the same time.
+```GO
+// NVIDIA Accelerators enables GPU related feature in koordlet. Only Nvidia GPUs supported.
+    Accelerators featuregate.Feature = "NvidiaAccelerator"
+// Accelerators enables heterogeneous GPU devices reporting.
+    Accelerators featuregate.Feature = "Accelerators"
+``` 
+
+### Device Infos Formate In File
```

Comment:
Please explain why we don't use CDI, AND how to compatible with different vendor when allocate